### PR TITLE
distributed checkpoint average script

### PR DESCRIPTION
**Pull request by yidong72**
# What does this PR do ?

a script to average distributed checkpoint files.

### CODE REVIEW

File: scripts/checkpoint_averaging/distributed_checkpoint_averaging.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    checkpoint_paths = []
+    for ckpt_dir in os.listdir(args.checkpoint_dir):
+        logging.info("Processing %s", ckpt_dir)
+        if ckpt_dir.endswith('0-last'):
+            continue
+        if args.steps is None:
+            checkpoint_paths.append(ckpt_dir)
+        else:
+            for step in steps:
```
[the remark this answers is not in the document]
Comment (titu1994) [reply scripts/checkpoint_averaging/distributed_checkpoint_averaging.py:80]: For step in args.steps?

File: scripts/checkpoint_averaging/distributed_checkpoint_averaging.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        else:
+            array_z = avg_weights[k][:]
+            array_z = array_z / n
+            avg_weights[k] = zarr.array(array_z, chunks=avg_weights[k].chunks, dtype=avg_weights[k].dtype)
+
+    # Save model
+    if args.steps is None:
+        ckpt_name = os.path.join(args.checkpoint_dir, args.name_prefix + '-averaged')
+    else:
+        steps_combined = '_'.join([str(x) for x in steps])
```
[the remark this answers is not in the document]
Comment (titu1994) [reply scripts/checkpoint_averaging/distributed_checkpoint_averaging.py:141]: args.steps?

**Review by titu1994 — COMMENTED** [review]
Minor comments overall looks great

File: scripts/checkpoint_averaging/distributed_checkpoint_averaging.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yidong72) [reply scripts/checkpoint_averaging/distributed_checkpoint_averaging.py]: fixed

File: scripts/checkpoint_averaging/distributed_checkpoint_averaging.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yidong72) [reply scripts/checkpoint_averaging/distributed_checkpoint_averaging.py]: fixed

**Review by titu1994 — APPROVED** [review]
Lgtm, thanks !

**Review by Kipok — APPROVED** [review]
Looks good on the high level - I don't know the details of the distributed checkpointing, so can't really check the logic thoroughly

**Pull request merged**